File: docs/docs/API-Reference/python-examples/api-flows-run/request-example-with-all-headers-and-parameters.py
import os

import requests

base = os.environ.get("LANGFLOW_URL") or os.environ.get("LANGFLOW_SERVER_URL", "")
flow_id = os.environ.get("FLOW_ID", "")
api_key = os.environ.get("LANGFLOW_API_KEY", "")

headers = {
    "Content-Type": "application/json",
    "accept": "application/json",
    "x-api-key": api_key,
}

payload = {
    "input_value": "Tell me a story",
    "input_type": "chat",
    "output_type": "chat",
    "output_component": "chat_output",
    "session_id": "chat-123",
}

response = requests.post(
    f"{base}/api/v1/run/{flow_id}?stream=false",
    headers=headers,
    json=payload,
    timeout=60,
)
response.raise_for_status()
print(response.text)

File: docs/versioned_docs/version-1.11.0/API-Reference/javascript-examples/api-flows-run/request-example-with-all-headers-and-parameters.js
const url = `${process.env.LANGFLOW_SERVER_URL ?? ""}/api/v1/run/${process.env.FLOW_ID ?? ""}?stream=true`;

const options = {
  method: 'POST',
  headers: {
    "Content-Type": `application/json`,
    "accept": `application/json`,
    "x-api-key": `${process.env.LANGFLOW_API_KEY ?? ""}`,
  },
  body: JSON.stringify({
  "input_value": "Tell me a story",
  "input_type": "chat",
  "output_type": "chat",
  "output_component": "chat_output",
  "session_id": "chat-123",
  "tweaks": {
    "component_id": {
      "parameter_name": "value"
    }
  }
}),
};

fetch(url, options)
  .then(async (response) => {
    if (!response.ok) {
      throw new Error(`HTTP ${response.status}`);
    }
    const text = await response.text();
    console.log(text);
  })
  .catch((error) => console.error(error));
